pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -260,16 +260,36 @@ func (p *Provider) createLaunchTemplate(ctx context.Context, options *amifamily.
 // This is done to help comply with AWS account policies that require explicitly setting that field to 'false'.
 // https://github.com/aws/karpenter/issues/3815
 func (p *Provider) generateNetworkInterface(options *amifamily.LaunchTemplate) []*ec2.LaunchTemplateInstanceNetworkInterfaceSpecificationRequest {
```

Comment:
Probably should rename the function to `generateNetworkInterfaces` (plural), since now we're generating multiple of them in a loop.